lib/msal-node/src/network/HttpClient.ts

```diff
@@ -274,3 +279,45 @@ const networkRequestViaHttps = <T>(
     });
 };
 
+/**
+ * Check if extra parsing is needed on the repsonse from the server
+ * @param statusCode {number} the status code of the response from the server
+ * @param statusMessage {string | undefined} the status message of the response from the server
+ * @param headers {Record<string, string>} the headers of the response from the server
+ * @param body {string} the body from the response of the server
+ * @returns {Object} JSON parsed body or error object
+ */
+const parseBody = (statusCode: number, statusMessage: string | undefined, headers: Record<string, string>, body: string) => {
+    /*
+     * Informational responses (100 – 199)
+     * Successful responses (200 – 299)
+     * Redirection messages (300 – 399)
+     * Client error responses (400 – 499)
+     * Server error responses (500 – 599)
+     */
+    
+    let parsedBody;
+    try {
+        parsedBody = JSON.parse(body);
+    } catch (error) {
```

Comment:
switch instead for readability?

Reply:
I would like ranges and errors in "Constants" and used here as these are pretty much industry standard.

Reply:
I had it as a switch originally... but there's only 2 options before the else. I think an if/else is better for readability. I will define those numbers as constants though.